compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/environment/ProjectPackageCache.java

```diff
@@ -98,4 +98,8 @@ private boolean isCurrentPackage(PackageOrg packageOrg, PackageName packageName,
         return descriptor.org().equals(packageOrg) && descriptor.name().equals(packageName) &&
                 descriptor.version().equals(version);
     }
+
+    @Override
+    @Deprecated
```

Comment:
Why do we have this deprecated? We should throw and unsupported operation exception instead.

Reply:
Throws an unsupported operation exception instread of having the deprecated tag